kconfig: Decode Kconfig sources as UTF-8 instead of as LC_CTYPE

SebastianBoe opened the pull request:
Decode Kconfig sources as UTF-8 instead of decoding them according to
the system locale (which might be ascii-only).

This resolves an issue that can be reproduced like this:

$ export LANG=C
$ echo '# Sebastian Bøe' >> ~/zephyr/Kconfig
$ build_any_zephyr_app

See the Python comment for details.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe committed:
kconfig: Decode Kconfig sources as UTF-8 instead of as LC_CTYPE

Decode Kconfig sources as UTF-8 instead of decoding them according to
the system locale (which might be ascii-only).

This resolves an issue that can be reproduced like this:

$ export LANG=C
$ echo '# Sebastian Bøe' >> ~/zephyr/Kconfig
$ build_any_zephyr_app

See the Python comment for details.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

carlescufi commented:
Looks good, is this present upstream and/or in our Kconfig fork?

carlescufi commented:
I think that Kconfiglib is also used by the documentation, should we change it there too?

SebastianBoe commented:
"Looks good, is this present upstream and/or in our Kconfig fork?"

This is a modification to a file that is maintained purely in Zephyr. It is not a part of any kconfiglib fork (I believe).

carlescufi commented:
>This is a modification to a file that is maintained purely in Zephyr. It is not a part of any kconfiglib fork (I believe).

You are right, I deleted my comment. Sorry for the noise

SebastianBoe commented:
"I think that Kconfiglib is also used by the documentation, should we change it there too?"
Really? I was not aware, I'll do some grepping to make sure this patch is applied everywhere it should be.

SebastianBoe committed:
doc: Kconfig: Decode Kconfig sources as UTF-8 instead of as LC_CTYPE

Decode Kconfig sources as UTF-8 instead of decoding them according to
the system locale (which might be ascii-only).

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe commented:
"I think that Kconfiglib is also used by the documentation, should we change it there too?"
Done.
